### ISSUE TITLE
Duplicate Service Name And application id is 0

### ISSUE THREAD

**Issue by candyleer** [apache/incubator-skywalking#1174]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [x] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?

___
### Bug
- Which version of SkyWalking, OS and JRE?

- Which company or project?

- What happen?
If possible, provide a way for reproducing the error. e.g. demo application, component version.

when I search in ui `service` page,I found each service has two record, and the `application_id` for one record is 0,I think this is a dirty data. such as
for `/hello/uvw`,the search  result in  index `service_name` are:
```json
{
    "took": 1,
    "timed_out": false,
    "_shards": {
        "total": 2,
        "successful": 2,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": 2,
        "max_score": 4.0748634,
        "hits": [
            {
                "_index": "service_name",
                "_type": "type",
                "_id": "29",
                "_score": 4.0748634,
                "_source": {
                    "service_name": "/hello/uvw",
                    "service_id": 29,
                    "src_span_type": 0,
                    "service_name_keyword": "/hello/uvw",
                    "application_id": -4
                }
            },
            {
                "_index": "service_name",
                "_type": "type",
                "_id": "-29",
                "_score": 3.9653742,
                "_source": {
                    "service_name": "/hello/uvw",
                    "service_id": -29,
                    "src_span_type": 0,
                    "service_name_keyword": "/hello/uvw",
                    "application_id": 0
                }
            }
        ]
    }
}
```
and trace is `cross thread`,like
![image](https://user-images.githubusercontent.com/11530760/39663704-a9b21874-50aa-11e8-94a3-bf66ec696734.png)

the reason I try to pinpoint maybe 
1. In `Agent`,`cross thread ` didn't have `parentApplicationInstanceId`,see code in `org.apache.skywalking.apm.agent.core.context.trace.TraceSegmentRef#transform`
```java
public TraceSegmentReference transform() {
        TraceSegmentReference.Builder refBuilder = TraceSegmentReference.newBuilder();
        if (SegmentRefType.CROSS_PROCESS.equals(type)) {
            refBuilder.setRefType(RefType.CrossProcess);
            refBuilder.setParentApplicationInstanceId(parentApplicationInstanceId);
            if (peerId == DictionaryUtil.nullValue()) {
                refBuilder.setNetworkAddress(peerHost);
            } else {
                refBuilder.setNetworkAddressId(peerId);
            }
        } else {
            refBuilder.setRefType(RefType.CrossThread);
        }

        refBuilder.setEntryApplicationInstanceId(entryApplicationInstanceId);
        refBuilder.setParentTraceSegmentId(traceSegmentId.transform());
        refBuilder.setParentSpanId(spanId);
        if (entryOperationId == DictionaryUtil.nullValue()) {
            refBuilder.setEntryServiceName(entryOperationName);
        } else {
            refBuilder.setEntryServiceId(entryOperationId);
        }
        if (parentOperationId == DictionaryUtil.nullValue()) {
            refBuilder.setParentServiceName(parentOperationName);
        } else {
            refBuilder.setParentServiceId(parentOperationId);
        }
        return refBuilder.build();
    }
```

but in `collector` code, this has been handled by this in `org.apache.skywalking.apm.collector.analysis.segment.parser.provider.parser.standardization.ReferenceIdExchanger#exchange`
```java
if (standardBuilder.getParentServiceId() == 0) {
            String parentServiceName = StringUtils.isNotEmpty(standardBuilder.getParentServiceName()) ? standardBuilder.getParentServiceName() : Const.DOMAIN_OPERATION_NAME;
            int parentServiceId = serviceNameService.getOrCreate(instanceCacheService.getApplicationId(standardBuilder.getParentApplicationInstanceId()), SpanType.Entry_VALUE, parentServiceName);

            if (parentServiceId == 0) {
                if (logger.isDebugEnabled()) {
                    int parentApplicationId = instanceCacheService.getApplicationId(standardBuilder.getParentApplicationInstanceId());
                    logger.debug("parent service name: {} from application id: {} exchange failed", parentServiceName, parentApplicationId);
                }
                return false;
            } else {
                standardBuilder.toBuilder();
                standardBuilder.setParentServiceId(parentServiceId);
                standardBuilder.setParentServiceName(Const.EMPTY_STRING);
            }
        }
```
in `standardBuilder.getParentApplicationInstanceId()` will always be 0 if in `cross thread` model,this maybe not as our expected.
so this will create a record which `application_id` is 0.
```java
int parentServiceId = serviceNameService.getOrCreate(instanceCacheService.getApplicationId(standardBuilder.getParentApplicationInstanceId()), SpanType.Entry_VALUE, parentServiceName);
```

#### maybe we should set `parentApplicationInstanceId` in `cross thread` in `Agent`.
___
### Requirement or improvement
- Please describe about your requirements or improvement suggestions.

**Comment by wu-sheng** [comment]
@candyleer Thanks for reporting this issue. The reason parentApplicationInstanceId is empty, because there is no need. For across thread, applicationInstanceId of parent is same as the current applicationInstanceId. 

Of course, this is a little harm to add this field into CROSS_THREAD REF. Just send one or two more bytes. So I think the easier way is to accept your proposal. Please send the pull request as soon as possible.

We are going to froze codes for preparing the next release.

**Comment by wu-sheng** [comment]
You only need to change `public TraceSegmentRef(ContextSnapshot snapshot)`, to set the parentApplicationInstanceId as the current instanceId.

**Comment by candyleer** [comment]
Yes,in `public TraceSegmentRef(ContextSnapshot snapshot)` ,the field is setted,but `transform()` missed this field ,I fixed in pr #1175

**Comment by wu-sheng** [comment]
FYI @peng-yongsheng This may be why there is service's applicationId is 0.
